### PR TITLE
Fix precision lost for floating-point qubos

### LINKED ISSUE #29: Default value of 1 for `scale_factor` means bad results bqms with non-integer biases
Though the parameter `scale_factor` is documented, the default behavior means casting all biases to `int` (not just rounding, but truncating the fractional part) which can be pathological for anything but very low precision problems.

A quick example (note I don't fully understand this as I would think it should produce random samples):
```python
In [21]: tabu_sampler.sample_ising({0: 0.4}, {}, num_reads=100).record.energy.mean()                                                                                                     
Out[21]: 0.3999999999999999
```

`scale_factor` should probably be set internally in the `sample` method, determined by the bqm range of biases. I can't think of any use-case that requires exposing it (though please mention it if there is one!)

### PR DESCRIPTION
**Pull request by randomir**
Closes #29.

### CODE REVIEW

File: tabu/sampler.py [hunk 1]
```diff
@@ -228,7 +235,8 @@ def _bqm_to_tabu_qubo(bqm):
         ud *= .5
         symm = ud + ud.T
         qubo = symm.tolist()
-        return qubo, varorder
+        maxbias = max(symm.max(), -symm.min())
```
Comment (arcondello) [inline tabu/sampler.py:238]: It might make sense to check that case that they are all naturally integers. Something like `(symm == symm.floor()).all()`.

Comment (randomir) [reply tabu/sampler.py:238]: Why? It's ok to get a float for maxbias

Comment (arcondello) [reply tabu/sampler.py:238]: Sorry, I annotated the wrong line. We won't need to scale in the case that they are all already integer.

Comment (randomir) [reply tabu/sampler.py:238]: Yes, that's a good idea.

**Review by hhtong — APPROVED** [review]

**Review by arcondello — APPROVED** [review]
I would still like to not scale in the case of integer biases. But don't view it as blocking.

See also https://github.com/dwavesystems/dimod/issues/699 which would make this simpler.

**Comment by randomir** [comment]
> I would still like to not scale in the case of integer biases. But don't view it as blocking.

I agree. It slipped my mind. Will add. Thanks.

**Comment by randomir** [comment]
Resolved with #67.

**Pull request closed without merging**